quickcheck-dynamic/src/Test/QuickCheck/DynamicLogic/Internal.hs

```diff
@@ -64,14 +64,30 @@ passTest = DynFormula . const $ Stop
 afterAny :: (Annotated s -> DynFormula s) -> DynFormula s
 afterAny f = DynFormula $ \n -> AfterAny $ \s -> unDynFormula (f s) n
 
+afterPolar
+  :: (Typeable a, Eq (Action s a), Show (Action s a))
+  => ActionWithPolarity s a
+  -> (Var a -> Annotated s -> DynFormula s)
+  -> DynFormula s
+afterPolar act f = DynFormula $ \n -> After act $ \x s -> unDynFormula (f x s) n
+
 -- | Given `f` must be `True` after /some/ action.
 -- `f` is passed the state resulting from executing the `Action`.
 after
   :: (Typeable a, Eq (Action s a), Show (Action s a))
   => Action s a
   -> (Var a -> Annotated s -> DynFormula s)
   -> DynFormula s
-after act f = DynFormula $ \n -> After act $ \x s -> unDynFormula (f x s) n
+after act f = afterPolar (ActionWithPolarity act PosPolarity) f
+
+-- | Given `f` must be `True` after /some/ action.
```

Comment:
This comment is exactly the same than the one for `after` which makes me doubt whether it's a copy/paste gone rogue. If the behavior is actually similar then perhaps a bit more explanation on why chose this one over the other would be helpful.